### PR TITLE
Adds guards patrolling together

### LINKED ISSUE #4562: Lumberjack exception
### Minecolonies version

Version: minecolonies-0.11.802-BETA-universal

Lumberjack threw several AI exceptions, see attached log file. I like the in-game error message, makes knowing when to report much easier. Game did not crash, and other citizens kept going.

### Logs: Latest.log and crashlog at https://gist.github.com/

- latest.log: 
[latest.log](https://github.com/ldtteam/minecolonies/files/4425842/latest.log)

- crashlog:

### PR DESCRIPTION
**Pull request by someaddons**
Closes #4562
Closes #
Closes #

# Changes proposed in this pull request:
Adds guards patrolling together, they choose a common patrol point and wait till everyone arrived or a timer runs out.
Simplified config settings for colony claims, there are now 3 settings: Max allowed range to claim, Min distance between townhalls and initial claim range.
Allow pirate ships to spawn on an ice surface too

Fixes:
Fixed citizen respawn when the last position he was at is blocked
Fixed supply camp reset command having no effect, since all it did to the advancement was +0
Fixed missing config description for one of the claim settings
Fixed lumberjack's trees causing an npe

Review please

### CODE REVIEW

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=5300938&bid=17350903)

File: src/api/java/com/minecolonies/api/colony/buildings/IGuardBuilding.java [hunk 1]
```diff
@@ -78,6 +77,8 @@ static boolean checkIfGuardShouldTakeDamage(AbstractEntityCitizen citizen, Playe
      */
     int getOffenceBonus();
 
+    void arrivedAtPatrolPoint(AbstractEntityCitizen guard);
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/colony/buildings/IGuardBuilding.java:85]: no jdoc?

File: src/api/java/com/minecolonies/api/colony/managers/interfaces/IBuildingManager.java [hunk 2]
```diff
@@ -180,6 +174,8 @@
      */
     BlockPos getBestHospital(AbstractEntityCitizen citizen);
 
+    BlockPos getRandomBuilding(Predicate<IBuilding> filterPredicate);
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/colony/managers/interfaces/IBuildingManager.java:183]: no jdoc?

File: src/api/java/com/minecolonies/api/colony/buildings/IGuardBuilding.java (hunk 1 above)
Comment (someaddons) [reply src/api/java/com/minecolonies/api/colony/buildings/IGuardBuilding.java:85]: added it^^

**Comment by marvin-bitterlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=5300938&bid=17350903)

**Review by marchermans — APPROVED** [review]

**Pull request merged**